<launch>

    <!-- 
        ###############################
        base params
        ###############################
    -->
    <arg name="simulation" default="false"/>

    <arg name="use_odom" default="true"/>
    <arg name="odom_port" default="/dev/ttyHoriokart-odom"/>

    <arg name="use_lidar" default="true"/>
    <arg name="front_rplidar_port" default="/dev/ttyHoriokart-frontlidar"/>
    <arg name="top_rplidar_port" default="/dev/ttyHoriokart-toplidar"/>

    <arg name="use_gps" default="true"/>
    <arg name="gps_port" default="/dev/ttyHoriokart-gps"/>

    <arg name="use_rs_d435i" default="false"/>
    <arg name="use_imu" default="false"/>

    <arg name="use_rs_d435" default="false"/>

    <arg name="use_ekf" default="false"/>

    <arg name="use_resource_pub" default="true"/>

    <!-- 
        ###############################
        hardware sensor drivers
        ###############################
    -->
        <!-- Odometry -->
        <group if="$(arg use_odom)">

            <node pkg="horiokart2021_sensors" type="serial_odom" name="serial_odom" output="screen">
                <param name="device_name" type="string" value="$(arg odom_port)" />
                <param name="odom_frame" type="string" value="odom" />
                <param name="base_frame" type="string" value="base_footprint" />
                <param name="odom_pub_rate" type="int" value="10" />

                <param name="inv_x" type="bool" value="true" />
                <param name="inv_y" type="bool" value="true" />
                <param name="inv_th" type="bool" value="false" />

                <param name="csv" type="bool" value="true" />
            </node>

        </group>


    <node name="pc_resource" pkg="horiokart2021_sensors" type="pc_resource_publisher.py" if="$(arg use_resource_pub)" />

</launch>
